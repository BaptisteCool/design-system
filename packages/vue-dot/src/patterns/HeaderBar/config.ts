export default {
	appBar: {},
	drawer: {
		temporary: true,
		fixed: true
	}
};
