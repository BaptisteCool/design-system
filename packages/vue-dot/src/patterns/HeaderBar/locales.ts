export default {
	account: 'Mon compte',
	logoBtn: {
		label: 'Accueil',
		alt: 'l\'Assurance Maladie'
	}
};
