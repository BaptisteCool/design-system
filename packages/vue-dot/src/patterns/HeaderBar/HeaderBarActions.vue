<template>
	<VList v-bind="options.list">
		<!-- list of optional actions -->
		<VListItem
			v-for="(item, index) in actionsList"
			:key="index"
			@click="clickAction(index)"
		>
			<VListItemContent>{{ item }}</VListItemContent>
		</VListItem>
		<!-- logout -->
		<VListItem
			v-if="loggedIn"
			@click="clickLogout"
		>
			<VIcon v-bind="options.logoutIcon">
				{{ mdiExitToApp }}
			</VIcon>
			{{ locales.logOut }}
		</VListItem>
	</VList>
</template>

<script lang="ts">
	import Vue from 'vue';
	import Component from 'vue-class-component';
	import header from './mixins/header';

	import customizable, { Options } from '../../mixins/customizable';
	import config from './config/HeaderBarActions';

	import { mdiExitToApp } from '@mdi/js';

	/** The profile button in the Header */
	@Component<HeaderBarActions>({
		mixins: [
			// Default configuration
			customizable(config),
			header
		]
	})
	export default class HeaderBarActions extends Vue {
		// Mixin computed data
		options!: Options;

		actionsList!: [string];
		mdiExitToApp = mdiExitToApp;
	}
</script>

<style scoped>
</style>
