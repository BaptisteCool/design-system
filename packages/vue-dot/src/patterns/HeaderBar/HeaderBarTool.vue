<template>
	<VToolbar
		:color="search === null ? 'primary' : 'transparant'"
		v-bind="options.toolbar"
		:height="$vuetify.breakpoint.smAndDown ? '36' : '48'"
	>
		<!-- search if search is not null or searchable -->
		<VTextField
			v-if="search !== null"
			v-bind="options.searchTextField"
			:dense="$vuetify.breakpoint.smAndDown"
			:placeholder="locales.search"
			@blur="search = null"
			@click:clear="search = null"
			@keydown.enter.prevent="emitSearch($event); search = null"
		/>
		<!-- toolbar components with 'back' icon action, breadcrumb, tabs and 'search' icon  -->
		<template v-else>
			<!-- back icon -->
			<VBtn
				v-if="back && breadcrumb"
				icon
				small
				dark
				@click.prevent="emitBack"
			>
				<VIcon v-bind="options.backIcon">
					{{ mdiArrowLeft }}
				</VIcon>
			</VBtn>
			<!-- breadcrumb -->
			<div
				v-if="breadcrumb"
				v-bind="options.breadcrumb"
			>
				{{ breadcrumb }}
			</div>
			<VDivider
				v-if="breadcrumb"
				v-bind="options.divider"
			/>
			<!-- select navigation in mobile mode -->
			<VSelect
				v-if="navigationList && navigationList.length > 0 && $vuetify.breakpoint.smAndDown"
				:value="value"
				v-bind="options.select"
				:items="navigationSelectItems"
				@change="$emit('input',$event)"
			>
				<template #selection="{item}">
					<div v-bind="options.selectedItem">
						{{ item.text }}
					</div>
				</template>
				<template #item="{item}">
					<div v-bind="options.selectItem">
						{{ item.text }}
					</div>
				</template>
			</VSelect>
			<!-- tabs if not in mobile -->
			<VTabs
				v-else-if="navigationList"
				:value="value"
				v-bind="options.tabs"
				@change="$emit('input',$event)"
			>
				<VTabsSlider />
				<VTab
					v-for="(item,i) in navigationList"
					:key="i"
					v-bind="options.tabItem"
				>
					{{ item }}
				</VTab>
			</VTabs>
			<VSpacer />
			<!-- search icon -->
			<VBtn
				v-if="searchable"
				v-bind="options.searchBtn"
				@click="search = ''"
			>
				<VIcon>{{ mdiMagnify }}</VIcon>
			</VBtn>
		</template>
	</VToolbar>
</template>

<script lang="ts">
	import Vue from 'vue';
	import Component from 'vue-class-component';
	import { mdiArrowLeft, mdiMagnify } from '@mdi/js';
	import header from './mixins/header';

	import customizable, { Options } from '../../mixins/customizable';
	import config from './config/HeaderBarTool';

	const Props = Vue.extend({
		props: {
			// tab number
			value: {
				type: Number,
				default: 0
			}
		}
	});

	@Component<HeaderBarTool>({
		mixins: [
			// Default configuration
			customizable(config),
			header
		]
	})
	export default class HeaderBarTool extends Props {
		// Mixin computed data
		options!: Options;
		// icons
		mdiArrowLeft = mdiArrowLeft;
		mdiMagnify = mdiMagnify;

		// mixins
		navigationList!: [string];

		// locales
		search: string | null = null;

		// compute the select items
		get navigationSelectItems() {
			return this.navigationList.map((item, index) => {
				return { text: item, value: index };
			});
		}
	}
</script>

<style scoped>
</style>
