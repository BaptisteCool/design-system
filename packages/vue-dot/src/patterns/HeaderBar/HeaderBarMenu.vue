<template>
	<VMenu
		v-if="buttonText || showIcon"
		v-model="menu"
		content-class="menu-active-el"
		transition="scale-transition"
		origin="top center"
		offset-y
		left
		@input="$emit('change', $event)"
	>
		<template
			#activator="{ on }"
		>
			<VBtn
				class="menu-el text-none px-4 text-right no-text-transform"
				:height="buttonHeight"
				:large="!$vuetify.breakpoint.smAndDown"
				text
				@click="$emit('click:menu')"
				v-on="on"
			>
				<VLayout
					v-if="buttonText"
					class="btn-text"
					column
				>
					<!-- agent firstname and lastname or account text    -->
					<div
						class="subtitle-1 text-truncate"
					>
						{{ buttonText }}
					</div>
					<!-- informations -->
					<div
						v-if="info"
						class="body-2 text-truncate"
					>
						{{ info.trim() }}
					</div>
				</VLayout>

				<!-- icon account
				when user is logged in and icon not hidden
				when mobile version and the user is not logged In
				-->
				<VIcon
					v-if="(!hideUserIcon && loggedIn && !$vuetify.breakpoint.smAndDown) || ($vuetify.breakpoint.smAndDown && !loggedIn)"
					color="grey darken-1"
					class="round-icon ml-1"
				>
					{{ userIcon }}
				</VIcon>
			</VBtn>
		</template>
		<VList
			v-if="!$vuetify.breakpoint.smAndDown"
			class="py-0 subtitle-1"
		>
			<!-- list of optional actions -->
			<VListItem
				v-for="(item, index) in actionsList"
				:key="index"
				@click="$emit('click:action', index)"
			>
				<VListItemContent>
					{{ item }}
				</VListItemContent>
			</VListItem>
			<!-- logout -->
			<VListItem
				v-if="loggedIn"
				@click="$emit('click:logout')"
			>
				<VIcon
					color="primary"
					class="mr-2"
				>
					{{ logoutIcon }}
				</VIcon>

				{{ locales.logOut }}
			</VListItem>
		</VList>
	</VMenu>
</template>

<script lang="ts">
	import Vue from 'vue';
	import locales from './locales';
	import Component from 'vue-class-component';

	import { mapActions, mapState } from 'vuex';

	import { mdiAccount, mdiExitToApp } from '@mdi/js';

	const Props = Vue.extend({
		props: {
			loggedIn: {
				type: Boolean,
				default: false
			},
			firstname: {
				type: String,
				default: null
			},
			hideUserIcon: {
				type: Boolean,
				default: false
			},
			accountText: {
				type: String,
				default: locales.account
			},
			lastname: {
				type: String,
				default: null
			},
			info: {
				type: String,
				default: null
			},
			actionsList: {
				type: [Array, Object],
				default: () => []
			}
		}
	});

	/** The profile button in the Header */
	@Component<HeaderMenu>({})
	export default class HeaderMenu extends Props {
		// Icons
		userIcon = mdiAccount;
		logoutIcon = mdiExitToApp;

		// locales
		locales = locales;

		menu = false;

		// compose the user display name
		get agent() {
			const firstname = typeof this.firstname === 'string' ? this.firstname.trim() : '';
			const lastname = typeof this.lastname === 'string' ? this.lastname.trim() : '';
			if (firstname.length > 0 || lastname.length > 0) {
				return `${firstname} ${lastname}`;
			}
			return null;
		}

		/** calcul the button height */
		get buttonHeight() {
			if (this.$vuetify.breakpoint.smAndDown) {
				return 32;
			} else {
				return this.info ? 71 : 55;
			}
		}

		// calcul the button text to display
		get buttonText(){
			if (this.$vuetify.breakpoint.smAndDown) return null;

			if( this.loggedIn ){
				return this.agent;
			} else if (this.accountText && this.accountText.length >0) {
				return this.accountText.trim();
			}

			return null;
		}

		/** calcul if show the icon */
		get showIcon(){
			return (!this.hideUserIcon && this.loggedIn && !this.$vuetify.breakpoint.smAndDown) || (this.$vuetify.breakpoint.smAndDown && !this.loggedIn);
		}
	}
</script>

<style lang="scss" scoped>

	.btn-text {
		max-width: 270px;
	}

	.menu-active-el {
		border-radius: 4px;
	}

	.menu-el {
		right: 15px;
		position: absolute;
		text-align: left;
	}

	.round-icon {
		border-radius: 50%;
	}

	@media only screen and (max-width: 600px) {
		.menu-el {
			position: relative;
			margin-top: 10px;
			right: 0;
		}
	}
</style>
