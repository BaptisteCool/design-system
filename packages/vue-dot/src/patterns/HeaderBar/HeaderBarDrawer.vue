<template>
	<VNavigationDrawer
		:value="value"
		v-bind="{...$attrs, ...options.drawer}"
		@input="toggleDrawer($event)"
	>
		<VListItem>
			<VListItemAvatar>
				<VIcon>{{ mdiAccount }}</VIcon>
			</VListItemAvatar>

			<VListItemContent>
				<VListItemTitle v-bind="options.userName">
					{{ firstname }} {{ lastname }}
				</VListItemTitle>
				<v-list-item-subtitle v-bind="options.userInfo">
					{{ info }}
				</v-list-item-subtitle>
			</VListItemContent>
		</VListItem>

		<VDivider />

		<HeaderBarActions
			:actions-list="actionsList"
			:logged-in="loggedIn"
			@click:action="clickAction($event); toggleDrawer(false)"
			@click:logout="clickLogout(); toggleDrawer(false)"
		/>
	</VNavigationDrawer>
</template>

<script lang="ts">
	import Vue from 'vue';
	import Component from 'vue-class-component';
	import { mdiAccount } from '@mdi/js';

	import customizable, { Options } from '../../mixins/customizable';
	import config from './config/HeaderBardrawer';

	import HeaderBarActions from './HeaderBarActions.vue';
	import header from './mixins/header';

	const Props = Vue.extend({
		props: {
			value: {
				type: Boolean,
				default: false
			}
		}
	});

	@Component<HeaderBarDrawer>({
			mixins: [
				// Default configuration
				customizable(config),
				header
			],
			components: { HeaderBarActions }
	})
	export default class HeaderBarDrawer extends Props {
		// Mixin computed data
		options!: Options;

		// icons
		mdiAccount = mdiAccount;

		// show the drawer or not
		toggleDrawer(value: boolean) {
			this.$emit('input', value);
		}

	}
</script>

<style lang="scss" scoped>
</style>
