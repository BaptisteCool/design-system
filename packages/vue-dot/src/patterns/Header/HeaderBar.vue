<template>
	<div>
		<VAppBar
			class="headerBar"
			v-bind="$attrs"
			:app="app"
			:elevation="4"
			cipped-left
			:height="$vuetify.breakpoint.xs ? '40' : '80'"
			:extension-height="$vuetify.breakpoint.xs ? '36' : '48'"
		>
			<VAppBarNavIcon
				v-if="showNavIcon"
				@click.stop="$emit('click:menu'); showActionList = !showActionList"
			/>

			<img
				:alt="locales.logoBtn.alt"
				src="../../assets/logo.svg"
				class="d-block"
				:height="$vuetify.breakpoint.xs ? '40' : '80'"
				@click="$emit('home')"
			>

			<VDivider
				vertical
				inset
				class="primary"
			/>

			<slot name="logo">
				<VToolbarTitle
					class="body-1 ml-md-4 ml-1"
				>
					{{ title }}
				</VToolbarTitle>
			</slot>
			<VSpacer />

			<HeaderBarMenu
				v-if="!$vuetify.breakpoint.xs"
				:account-text="accountText"
				:logged-in="loggedIn"
				:info="info"
				:hide-user-icon="hideUserIcon"
				:actions-list="actionsList"
				:firstname="firstname"
				:lastname="lastname"
				@click:menu="$emit('click:menu')"
				@click:action="$emit('click:action', $event)"
				@click:logout="$emit('click:logout')"
			/>

			<template
				v-if="navigationList || breadcrumb || search"
				#extension
			>
				<VToolbar
					color="primary"
					class="px-4"
					:height="$vuetify.breakpoint.xs ? '36' : '48'"
				>
					<VBtn
						v-if="back && breadcrumb"
						icon
						small
						dark
						@click.prevent="$emit('return')"
					>
						<VIcon
							color="white"
						>
							{{ mdiArrowLeft }}
						</VIcon>
					</VBtn>
					<VCol
						v-if="breadcrumb"
						class="white--text text-no-wrap"
					>
						{{ breadcrumb }}
					</VCol>
					<VDivider
						v-if="breadcrumb"
						vertical
						inset
						class="white mx-sm-4 mr-2 mb-2"
					/>
					<VSelect
						v-if="navigationList && $vuetify.breakpoint.xs"
						v-model="tab"
						dark
						hide-details
						:items="NavSelectItems"
					/>
					<VTabs
						v-else-if="navigationList"
						v-model="tab"
						:show-arrows="true"
						dark
						background-color="transparent"
					>
						<VTabsSlider />

						<VTab
							v-for="(item,i) in navigationList"
							:key="i"
						>
							{{ item }}
						</VTab>
					</VTabs>
					<VSpacer />
					<VBtn
						icon
						small
						color="white"
					>
						<VIcon>
							{{ mdiMagnify }}
						</VIcon>
					</VBtn>
				</VToolbar>
			</template>
		</VAppBar>
		<HeaderBarDrawer
			v-model="showActionList"
			:app="app"
			temporary
			fixed
			:clipped="clipped"
			:info="info"
			:actions-list="actionsList"
			:firstname="firstname"
			:lastname="lastname"
			@click:action="$emit('click:action', $event)"
			@click:logout="$emit('click:logout')"
		/>
	</div>
</template>

<script lang="ts">
	import Vue from 'vue';
	import locales from './locales';
	import Component from 'vue-class-component';
	import { mdiArrowLeft, mdiMagnify, mdiAccount, mdiExitToApp } from '@mdi/js';

	import HeaderBarMenu from './HeaderBarMenu.vue';
	import HeaderBarDrawer from './HeaderBarDrawer.vue';

	import { mapState } from 'vuex';

	const Props = Vue.extend({
		props: {
			value: {
				type: Number,
				default: 0
			},
			logo: {
				type: String || Object,
				default: null
			},
			title: {
				type: String,
				required: true
			},
			breadcrumb: {
				type: String,
				default: null
			},
			app: {
				type: Boolean,
				default: true
			},
			back: {
				type: Boolean,
				default: false
			},
			search: {
				type: Boolean,
				default: false
			},
			navigationList: {
				type: Array,
				default: null
			},
			loggedIn: {
				type: Boolean,
				default: false
			},
			firstname: {
				type: String,
				default: null
			},
			hideUserIcon: {
				type: Boolean,
				default: false
			},
			accountText: {
				type: String,
				default: locales.account
			},
			lastname: {
				type: String,
				default: null
			},
			info: {
				type: String,
				default: null
			},
			actionsList: {
				type: [Array, Object],
				default: () => []
			}
		}
	});

/** The Header of the application */
@Component<HeaderBar>({
  components: {
    HeaderBarMenu, HeaderBarDrawer
  }
})
	export default class HeaderBar extends Props {
		locales = locales;
		mdiArrowLeft = mdiArrowLeft;
		mdiMagnify = mdiMagnify;
		mdiAccount = mdiAccount;
		mdiExitToApp = mdiExitToApp;
		showActionList = false;
		clipped = true;

		tab = null;

		get showNavIcon() {
			const show = this.$vuetify.breakpoint.xs && (this.actionsList.length > 0 || this.loggedIn);
			if (!show) {
			this.showActionList = false;
			}
			return show;
		}

		get NavSelectItems() {
			return this.navigationList.map((item, index) => {
			return { text: item, value: index };
			});
		}
	}
</script>

<style lang="scss" scoped>
.headerBar ::v-deep .v-toolbar__extension {
  padding: 4px 0px;
}

.headerBar ::v-deep .v-toolbar__content {
  padding: 4px 16px;
}
</style>
