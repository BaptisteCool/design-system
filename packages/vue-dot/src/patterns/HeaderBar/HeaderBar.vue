<template>
	<div>
		<VAppBar
			class="header-bar"
			v-bind="$attrs"
			:app="app"
			:elevation="4"
			:height="$vuetify.breakpoint.smAndDown ? '40' : '80'"
			:extension-height="$vuetify.breakpoint.smAndDown ? '36' : '48'"
		>
			<!-- icon menu for mobile -->
			<VAppBarNavIcon
				v-if="showNavDrawer"
				@click.stop="$emit('click:menu'); showActionList = !showActionList"
			/>
			<!-- logo CNAM -->
			<img
				:alt="locales.logoBtn.alt"
				src="../../assets/logo.svg"
				class="d-block"
				:height="$vuetify.breakpoint.smAndDown ? '40' : '80'"
				@click="$emit('home')"
			>

			<VDivider
				vertical
				inset
				class="primary"
			/>
			<!-- title slot -->
			<slot name="title">
				<VToolbarTitle
					class="body-1 ml-md-4 ml-1 header-title"
				>
					{{ title }}
				</VToolbarTitle>
			</slot>
			<VSpacer />
			<!-- menu right -->
			<HeaderBarMenu
				:account-text="accountText"
				:logged-in="loggedIn"
				:info="info"
				:hide-user-icon="hideUserIcon"
				:actions-list="actionsList"
				:firstname="firstname"
				:lastname="lastname"
				@click:menu="$emit('click:menu')"
				@click:action="$emit('click:action', $event)"
				@click:logout="$emit('click:logout')"
			/>
			<!-- tool bar in extension -->
			<template
				v-if="navigationList || breadcrumb || searchable"
				#extension
			>
				<HeaderBarTool
					:value="value"
					:breadcrumb="breadcrumb"
					:back="back"
					:searchable="searchable"
					:navigation-list="navigationList"
					@input="$emit('input',$event)"
					@back="$emit('back')"
					@search="$emit('search', $event)"
				/>
			</template>
		</VAppBar>
		<!-- drawer -->
		<HeaderBarDrawer
			v-if="showNavDrawer"
			v-model="showActionList"
			:app="app"
			temporary
			fixed
			:info="info"
			:actions-list="actionsList"
			:firstname="firstname"
			:lastname="lastname"
			@click:action="$emit('click:action', $event)"
			@click:logout="$emit('click:logout')"
		/>
	</div>
</template>

<script lang="ts">
	import Vue from 'vue';
	import locales from './locales';
	import Component from 'vue-class-component';

	import HeaderBarMenu from './HeaderBarMenu.vue';
	import HeaderBarDrawer from './HeaderBarDrawer.vue';
	import HeaderBarTool from './HeaderBarTool.vue';

	import { mapState } from 'vuex';
	import { watch } from 'fs';
	import { log } from '../../../../cli-helpers';

	const Props = Vue.extend({
		props: {
			// tab number
			value: {
				type: Number,
				default: 0
			},
			logo: {
				type: String || Object,
				default: null
			},
			title: {
				type: String,
				required: true
			},
			breadcrumb: {
				type: String,
				default: null
			},
			app: {
				type: Boolean,
				default: true
			},
			back: {
				type: Boolean,
				default: false
			},
			searchable: {
				type: Boolean,
				default: false
			},
			navigationList: {
				type: Array,
				default: null
			},
			loggedIn: {
				type: Boolean,
				default: false
			},
			firstname: {
				type: String,
				default: null
			},
			hideUserIcon: {
				type: Boolean,
				default: false
			},
			accountText: {
				type: String,
				default: locales.account
			},
			lastname: {
				type: String,
				default: null
			},
			info: {
				type: String,
				default: null
			},
			actionsList: {
				type: [Array, String],
				default: () => []
			}
		}
	});

	/** The Header of the application */
	@Component<HeaderBar>({
		components: {
			HeaderBarMenu, HeaderBarDrawer, HeaderBarTool
		}
	})
	export default class HeaderBar extends Props {
		locales = locales;
		showActionList = false;

		/**
		 * calculate if the drawer can be viewed in mobile:
		 * logged in user
		 * or actionsList is not empty
		 */
		get showNavDrawer() {
			const show = this.$vuetify.breakpoint.smAndDown && (this.actionsList.length > 0 || this.loggedIn);
			if (!show) {
			this.showActionList = false;
			}
			return show;
		}

	}
</script>

<style lang="scss" scoped>

.header-title {
	max-width: 300px;
}

.header-bar ::v-deep .v-toolbar__extension {
  padding: 4px 0px;
}

.header-bar ::v-deep .v-toolbar__content {
  padding: 4px 16px;
}
</style>
