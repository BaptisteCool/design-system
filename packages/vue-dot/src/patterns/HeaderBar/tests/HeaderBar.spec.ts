import Vue from 'vue';
import { Wrapper } from '@vue/test-utils';

import { mountComponent } from '@/tests';
import html from '@/tests/html';

import HeaderBar from '../HeaderBar.vue';

let wrapper: Wrapper<Vue>;

// Tests
describe('HeaderBar test', () => {
	it('renders correctly with all option', () => {
		// Mount component
		wrapper = mountComponent(HeaderBar, {
			propsData: {
				value : 2,
				app:true,
				title:'Titre long test',
				navigationList:'[\'onglet1\', \'onglet2\', \'onglet3\', \'onglet4\', \'onglet5\']',
				actionsList:'[\'test1\', \'test2\']',
				breadcrumb:'sous projet',
				back: true,
				searchable: true,
				hideUserIcon:false,
				loggedIn:true,
				firstname:'Jean philippe baptiste gezg  hre hh',
				lastname:'Coulon',
				info:'info supp fz g g egerggg g'
			}
		});

		expect(html(wrapper)).toMatchSnapshot();
	});

});
