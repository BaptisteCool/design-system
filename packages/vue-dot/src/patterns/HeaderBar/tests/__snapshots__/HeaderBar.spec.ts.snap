// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBar test renders correctly with all option 1`] = `
<div>
  <vappbar-stub elevation="4" height="80" tag="header" tile="true" extensionheight="48" src="" value="true" app="true" class="header-bar">
    <!----> <img alt="l'Assurance Maladie" src="../../assets/logo.svg" height="80" class="d-block">
    <vdivider-stub inset="true" vertical="true" class="primary"></vdivider-stub>
    <vtoolbartitle-stub class="body-1 ml-md-4 ml-1 header-title">
      Titre long test
    </vtoolbartitle-stub>
    <vspacer-stub></vspacer-stub>
    <headerbarmenu-stub loggedin="true" firstname="Jean philippe baptiste gezg  hre hh" accounttext="Mon compte" lastname="Coulon" info="info supp fz g g egerggg g" actionslist="['test1', 'test2']"></headerbarmenu-stub>
  </vappbar-stub>
  <!---->
</div>
`;
